include/cuco/detail/utils.cuh

```diff
@@ -122,13 +125,19 @@ __host__ __device__ constexpr SizeType sanitize_hash(HashType hash) noexcept
  *
  * @tparam SizeType The target type
  * @tparam HashType The input type
+ * @tparam CG Cooperative group type
  *
  * @return Converted hash value
  */
-template <typename SizeType, typename HashType>
-__host__ __device__ constexpr SizeType sanitize_hash(HashType hash, std::uint32_t cg_rank) noexcept
+template <typename SizeType, typename HashType, typename CG>
+__host__ __device__ constexpr SizeType sanitize_hash(HashType hash, CG group) noexcept
 {
-  return sanitize_hash<SizeType>(sanitize_hash<SizeType>(hash) + cg_rank);
+  auto const base_hash = sanitize_hash<SizeType>(hash);
+  auto const max_size  = cuda::std::numeric_limits<SizeType>::max();
+  auto const cg_rank   = static_cast<SizeType>(group.thread_rank());
+
+  if (base_hash > (max_size - cg_rank)) return cg_rank - (max_size - base_hash);
```

Comment:
I might be confused but `cg_rank - (max_size - base_hash)` can easily become a negative number which is not what we are looking for as the return value of this function. Am I missing something?

Reply:
Resolved during offline discussion.